### PR TITLE
fix/Add conditional if response received from graph api v1 was empty

### LINKED ISSUE #5546: [CAL-378] MS Teams link not showing in the confirmation email
> From [SyncLinear.com](https://synclinear.com) | [CAL-378](https://linear.app/calcom/issue/CAL-378/ms-teams-link-not-showing-in-the-confirmation-email)

### PR DESCRIPTION
**Pull request by alannnc**
## What does this PR do?

Looks like due to some unknown behaviour MSTeams createMeeting return 204 

- Add conditional to throw error if response from msteam graph api was empty and 2XX.
- console.error on handleErrorRaw to logs which services shouldn't throw 204.

Fixes #5546 

**Environment**: Staging(main branch) / Production

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: packages/app-store/office365video/lib/VideoApiAdapter.ts [hunk 1]
```diff
@@ -153,6 +153,10 @@ const TeamsVideoApiAdapter = (credential: CredentialPayload): VideoApiAdapter =>
 
       const resultObject = JSON.parse(resultString);
 
+      if (!resultObject.id || !resultObject.joinUrl) {
+        throw new HttpError({ statusCode: 500, message: "Error creating MS Teams meeting" });
+      }
+
```
Comment (alannnc) [inline packages/app-store/office365video/lib/VideoApiAdapter.ts:159]: if resultObject returned 204 status it's body will be {}, so doing resultObject.id will return as undefined and not returning error to main function createMeeting()

File: packages/lib/errors.ts [hunk 2]
```diff
@@ -30,6 +30,7 @@ export async function handleErrorsJson<Type>(response: Response): Promise<Type>
 
 export function handleErrorsRaw(response: Response) {
   if (response.status === 204) {
+    console.error({ response });
```
Comment (alannnc) [inline packages/lib/errors.ts:33]: This can be omitted but I would like to take a deep dive to see in what scenarios we return 204 {}

**Review by Udit-takkar — COMMENTED** [review]
this is the resultString i am getting in /videoApiAdapter

{"error":{"code":"Forbidden","message":"An error has occurred.","innerError":{"date":"2022-11-18T11:26:13","request-id":"2bbcbd13-7bc4-4a64-af78","client-request-id": <client_id_here> }}}

my outlook email comes under work/school mail. maybe some permissions are required or this app has to be approved by the organisation to use. Although we are giving the required permission OnlineMeetings.ReadWrite. don't know what's causing this.

**Comment by alannnc** [comment]
> this is the resultString i am getting in /videoApiAdapter
> 
> {"error":{"code":"Forbidden","message":"An error has occurred.","innerError":{"date":"2022-11-18T11:26:13","request-id":"2bbcbd13-7bc4-4a64-af78","client-request-id": <client_id_here> }}}
>

I think Forbidden responses are Unauthorised calls to API and we handle that with an Error in handleErrorsRaw()
if (!response.ok && response.status < 200 && response.status >= 300) 
In this case there wasn't an error but the response was empty.

Still a guess for me but we gain some visibility in logs if it happens again.

**Review by joeauyeung — APPROVED** [review]
LGTM, we just need to remember to check the logs once this is out.

**Pull request merged**